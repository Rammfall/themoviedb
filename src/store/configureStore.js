import { createLogicMiddleware } from 'redux-logic'
import { createStore, applyMiddleware } from 'redux'

import logics from './logics'
import rootReducer from './rootReducer'

const bindMiddleware = (middleware) => {
  if (process.env.NODE_ENV !== 'production') {
    // eslint-disable-next-line
    const { composeWithDevTools } = require('redux-devtools-extension')

    return composeWithDevTools(applyMiddleware(...middleware))
  }

  return applyMiddleware(...middleware)
}

const configureStore = (initialState = {}) => {
  // Add dependencies to pass them to logic functions
  const dependencies = {}

  const logicMiddleware = createLogicMiddleware(logics, dependencies)
  const store = createStore(rootReducer, initialState, bindMiddleware([logicMiddleware]))

  return store
}

export default configureStore
