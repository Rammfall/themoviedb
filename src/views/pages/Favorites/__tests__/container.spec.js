import React from 'react'
import { shallow } from 'enzyme'
import configureStore from 'redux-mock-store'

import diveTo from 'TestUtils/diveToEnzyme'

import FavoritesPage, { FavoritesPageContainer } from '../container'

jest.mock('react-intl', () => ({
  injectIntl: jest.fn(component => component)
}))

jest.mock('Store/concepts/movies/selectors', () => ({
  getFavoritesMoviesSelector: jest.fn(() => ({
    isEmpty: false
  }))
}))

jest.mock('Store/concepts/account/selectors', () => ({
  userIdSelector: jest.fn(() => 43)
}))

jest.mock('Store/concepts/data/selectors', () => ({
  loadingSelector: jest.fn()
}))

jest.mock('react-router-dom', () => ({
  withRouter: component => component
}))

describe('FavoritesPage', () => {
  const store = configureStore()()
  const props = {
    location: {
      search: ''
    },
    userId: 3,
    movies: {
      isEmpty: false
    },
    intl: {
      formatMessage: jest.fn(() => 'test')
    }
  }
  const wrapper = shallow(
    <FavoritesPage
      store={store}
      {...props}
    />
  )
  const container = diveTo(wrapper, FavoritesPageContainer)

  it('matches snapshot', () => {
    expect(container).toMatchSnapshot()
  })

  describe('componentDidUpdate()', () => {
    const anotherContainer = diveTo(wrapper, FavoritesPageContainer)
    jest.spyOn(anotherContainer.instance(), 'getData')
    anotherContainer.setProps({ userId: 2 })
    anotherContainer.setProps({ userId: undefined })
    anotherContainer.setProps({ location: { search: 'test' } })
    anotherContainer.setProps({ location: { search: 'test' } })

    expect(anotherContainer).toMatchSnapshot()
  })
})
