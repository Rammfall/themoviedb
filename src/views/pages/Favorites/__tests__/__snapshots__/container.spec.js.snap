// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<ListContent
  isEmpty={false}
  isLoading={true}
  list={<Memo(Connect(FavoriteMovies)) />}
  title="test"
  titleEmptyState="test"
/>
`;

exports[`FavoritesPage matches snapshot 1`] = `
<ListContent
  isEmpty={false}
  isLoading={true}
  list={<Memo(Connect(FavoriteMovies)) />}
  title="test"
  titleEmptyState="test"
/>
`;
